src/arduino/Adafruit_USBD_Device.cpp

```diff
@@ -316,8 +321,11 @@ uint16_t const *Adafruit_USBD_Device::descriptor_string_cb(uint8_t index,
     break;
 
   case STRID_SERIAL:
-    chr_count = getSerialDescriptor(_desc_str);
-    break;
+    if (!_desc_str_arr[STRID_SERIAL]) {
+      chr_count = getSerialDescriptor(_desc_str);
```

Comment:
this code should be moved insided getSerialDescriptor()

Reply:
Updated. Hope that's what you were looking for.